src/components/dapps/ModalCreateDapps.vue

```diff
@@ -284,32 +310,310 @@
+    });
+
+    const extensionFile = ['.contract', '.json'];
+
+    const onDropFile = (fileState: FileState) => {
+      setWasmFile(fileState);
+    };
+
+    const { wasm, isWasmValid } = useWasm(
+      wasmFromFile.value,
+      isWasmFromFileValid
+    );
+
+    const { onSend } = useSendTx();
+
+    const upload = async () => {
+      if (
+        formData.projectName === '' ||
+        toAddress.value === '' ||
+        wasm.value === null ||
+        !isWasmValid
+      ) {
+        store.dispatch(ActionTypes.SHOW_ALERT_MSG, {
+          msg: `Please check fields again`,
+          alertType: 'error',
+        });
+        return;
+      }
+
+      const abiData = abi.value as Abi | AnyJson;
+
+      console.log('s', abiData);
+      console.log('w', wasm.value);
+
+      let uploadTx: SubmittableExtrinsic<'promise'> | null = null;
+
+      try {
+        const code = new CodePromise(apiPromise, abiData, wasm.value);
+
+        //should be changable
+        // const unit_d = 3;
+        // const decimal = 12;
+        // const endowment = bnToBn(
+        //   plasmUtils.reduceDenomToBalance(27, unit_d, decimal)
+        // );
+        // const weight = new BN(200000000000);
+        ///
+        console.log('endowment', formData.endowment);
+        console.log('weight', formData.weight);
+
+        console.log('code', code);
+        console.log('method', code.tx);
+
+        const constructorIndex = 0;
+        const params = abi?.value?.constructors[constructorIndex].args;
+        console.log('params', params);
+
+        const arrValues = getParamValues(abi.value?.registry, params);
+        console.log('values', arrValues);
+
+        uploadTx =
+          code &&
+          abi.value?.constructors[constructorIndex].method &&
+          formData.endowment
+            ? code.tx[abi.value?.constructors[constructorIndex].method](
+                {
+                  gasLimit: formData.weight,
+                  value: formData.endowment,
+                },
+                ...arrValues
+              )
+            : null;
+      } catch (e) {
+        const error = (e as Error).message;
+        console.error(error);
+        store.dispatch(ActionTypes.SHOW_ALERT_MSG, {
+          msg: error,
+          alertType: 'error',
+        });
+        return;
+      }
+
+      console.log('uploadTx', uploadTx);
+
+      //send from txButton
+      const _onFailed = (result: SubmittableResult | null) => {
+        console.error('_onFailed', result);
+        store.commit(MutationTypes.SET_LOADING, false);
+        store.dispatch(ActionTypes.SHOW_ALERT_MSG, {
+          msg: result,
+          alertType: 'error',
+        });
+      };
+
+      const _onStart = () => {
+        console.log('_onStart');
+        store.commit(MutationTypes.SET_LOADING, true);
+      };
+
+      const _onSuccess = (result: CodeSubmittableResult) => {
+        console.log('_onSuccess', result);
+
+        const codeHash = result.blueprint?.codeHash;
```

Comment:
The contract hash is generated and stored locally, but it is not deployed to the chain.

Reply:
how can I verify whether this contract is deployed to the chain? because currently, it seems work similar with polkadot explorer

Reply:
I tried deploying the contract from plasm apps vue and access the address from Polkadot-js apps portal. It fails to load the contract. If you can access the contract from both applications, please let me know! Because it could be an issue with my local environment (perhaps related to chain types?)

Reply:
What about adding a unit test for checking API operations? We can either use mock data or real async calls to the node.
I'm working on a GitHub action for loading a local development node as part of the workflow (still WIP). https://github.com/hoonsubin/build-substrate-action
Having a unit test can definitely help development

Reply:
@sirius651 I want to clarify my statement above. I was able to check that the contract does deploy to the blockchain. But (outside of the issue of not being able to enter the constructor params) I can't sign and deploy with the address that I choose from the dropdown.

(in the contract, it says the creator is Alice, but I tried to deploy with a different account)
![Screen Shot 2021-06-08 at 2 18 07 PM](https://user-images.githubusercontent.com/40356749/121127691-200e6b00-c865-11eb-9feb-408bf7ffccc8.png)
![Screen Shot 2021-06-08 at 2 17 50 PM](https://user-images.githubusercontent.com/40356749/121127698-21d82e80-c865-11eb-9917-8a2cd2011bd2.png)

Reply:
> What about adding a unit test for checking API operations? We can either use mock data or real async calls to the node.
> I'm working on a GitHub action for loading a local development node as part of the workflow (still WIP). https://github.com/hoonsubin/build-substrate-action
> Having a unit test can definitely help development

yeah, I totally agree on that. I'll set up unit test when I have time

Reply:
> @sirius651 I want to fix my statement above. I was able to check that the contract does deploy to the blockchain. But (outside of the issue of not being able to enter the constructor params) I can't sign and deploy with the address that I choose from the dropdown. So I think that was the error that I saw.
> 

I found out that is bug. will be fixed on new branch : feature/deploy-contract